src/vs/editor/contrib/colorPicker/browser/defaultDocumentColorProvider.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { Color, RGBA } from 'vs/base/common/color';
+import { ITextModel } from 'vs/editor/common/model';
+import { DocumentColorProvider, IColor, IColorInformation, IColorPresentation, ProviderResult } from 'vs/editor/common/languages';
+import { EditorWorkerClient } from 'vs/editor/browser/services/editorWorkerService';
+import { IModelService } from 'vs/editor/common/services/model';
+import { ILanguageConfigurationService } from 'vs/editor/common/languages/languageConfigurationRegistry';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { ILanguageFeaturesService } from 'vs/editor/common/services/languageFeatures';
+import { registerEditorFeature } from 'vs/editor/common/editorFeatures';
+
+export class DefaultDocumentColorProvider implements DocumentColorProvider {
+
+	private _editorWorkerClient: EditorWorkerClient | null;
+
+	constructor(
+		modelService: IModelService,
+		languageConfigurationService: ILanguageConfigurationService,
+	) {
+		this._editorWorkerClient = new EditorWorkerClient(modelService, false, 'editorWorkerService', languageConfigurationService);
+	}
+
+	provideDocumentColors(model: ITextModel, _token: CancellationToken): ProviderResult<IColorInformation[]> {
+		return this._editorWorkerClient?.computeDefaultDocumentColors(model.uri).then((documentColors) => {
```

Comment:
Isn't `_editorWorkerClient` always set and the `?` can be removed?

Reply:
My bad, thanks for pointing it out